src/hotspot/share/gc/g1/g1UncommitRegionTask.cpp

```diff
@@ -129,7 +130,24 @@ void G1UncommitRegionTask::execute() {
   } else {
     // Nothing more to do, change state and report a summary.
     set_active(false);
+    _active_sem.signal();
     report_summary();
     clear_summary();
   }
 }
+
+void G1UncommitRegionTask::_wait_if_active() {
+  assert(Thread::current_or_null() != NULL, "no current thread");
+  assert(!Thread::current()->is_ConcurrentGC_thread(), "deadlock prevention");
+
+  while (_active) {
```

Comment:
Is this cycle safe against infinite loop? _active is not atomic or volatile.

Reply:
It is true I would normally make it `volatile` (it is atomic as it is small enough), it was forgotten as other code around is using it without `volatile`. But you are right this is a different use case of `_active` where cachine may have some effect.
OTOH for my defense the `volatile` should not be required due to a memory barrier there: https://stackoverflow.com/questions/4476446/can-a-c-c-compiler-legally-cache-a-variable-in-a-register-across-a-pthread-lib